### PR TITLE
added third java lab

### PR DESCRIPTION
**Pull request by Sasha-Sobran**

### CODE REVIEW

File: src/test/java/ua/lviv/iot/algo/part1/lab3/AbstractLaptopManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.jupiter.api.DisplayName;
+
+import static org.junit.Assert.assertEquals;
+
+public class AbstractLaptopManagerTest {
+    AbstractLaptopManager laptopsManager;
+    AbstractLaptop laptop1;
+    AbstractLaptop laptop2;
+    AbstractLaptop laptop3;
+    AbstractLaptop laptop4;
```
Comment (MarkoYaminsky) [inline src/test/java/ua/lviv/iot/algo/part1/lab3/AbstractLaptopManagerTest.java:14]: Why adding 4 fields which are temporary, needed only to fill laptopsManager and not tested in any way?

**Review by MarkoYaminsky — CHANGES_REQUESTED** [review]
Everything's fine, but those 4 fields are redundant

**Review by MarkoYaminsky — APPROVED** [review]

File: src/main/java/ua/lviv/iot/algo/part1/lab3/DualLogger.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class DualLogger implements Logger {
+    private String ultrabookExceptionMessage;
+
+    public String getultrabookExceptionMessage() {
+        return ultrabookExceptionMessage;
+    }
+
+    @Override
+    public void replaceBatteryInUltrabook(String replaceBatteryInUltrabookException) {
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/DualLogger.java:11]:
```suggestion
    public void log(String logMessage) {
```

File: src/main/java/ua/lviv/iot/algo/part1/lab3/DualLogger.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package ua.lviv.iot.algo.part1.lab3;
+
+public class DualLogger implements Logger {
+    private String ultrabookExceptionMessage;
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/DualLogger.java:4]: Your logger should not know anything about what calles it. Remove ultrabook from names here

File: src/main/java/ua/lviv/iot/algo/part1/lab3/GamingLaptop.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 public class GamingLaptop extends AbstractLaptop {
     private int numberOfFans;
     private String graphicsProcessor;
+    private  Logger logger;
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/GamingLaptop.java:13]:
```suggestion
    private Logger logger;
```

File: src/main/java/ua/lviv/iot/algo/part1/lab3/Logger.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package ua.lviv.iot.algo.part1.lab3;
+
+public interface Logger {
+    void replaceBatteryInUltrabook(String replaceBatteryInUltrabookException);
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/Logger.java:4]: Same here

File: src/main/java/ua/lviv/iot/algo/part1/lab3/Netbook.java [hunk 6]
```diff
@@ -18,11 +19,14 @@ public Netbook(final String model,
         final int batteryLife,
         final int currentBatteryLife,
         final String operatingSystem,
-        final String webCam
+        final String webCam,
+        final Logger logger
+
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/Netbook.java:24]:
```suggestion
```

File: src/main/java/ua/lviv/iot/algo/part1/lab3/Ultrabook.java [hunk 7]
```diff
@@ -1,16 +1,19 @@
 package ua.lviv.iot.algo.part1.lab3;
 
+import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.ToString;
 
+@Getter
```
Comment (BogdanDidukh2003) [inline src/main/java/ua/lviv/iot/algo/part1/lab3/Ultrabook.java]: Do you need it?

**Review by BogdanDidukh2003 — APPROVED** [review]

**Pull request merged**